.github/workflows/deploy.yml

```diff
@@ -0,0 +1,60 @@
+# name: Deploy CDK Stack
+# on:
+#   workflow_dispatch:
+#     inputs:
+#       environment:
+#         description: "Select deployment environment"
+#         required: true
+#         default: "develop"
+#         type: choice
+#         options:
+#           - develop
+#           - staging
+#           - production
+name: Test CDK Deploy
+on:
+  push:
+    branches:
+      - add/s3-recovery-stack  
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: develop
+
+    permissions:
+      pull-requests: write
+      id-token: write
+      contents: read
+
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set environment variables
+        run: |
+          echo "SOURCE_BUCKET_NAME=${{ vars.SOURCE_BUCKET_NAME }}" >> .env
+          echo "DESTINATION_BUCKET_NAME=${{ vars.DESTINATION_BUCKET_NAME }}" >> .env
+          echo "ALLOW_BATCH_REPLICATION=${{ vars.ALLOW_BATCH_REPLICATION }}" >> .env
+          echo "Env"=${{ vars.ENV }} >> .env
+          cat .env
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.DEPLOYMENT_ROLE_ARN }}
```

Comment:
To avoid searching for the role in the future, can we construct the ARN so that sensitive variables (e.g., the account number) are stored in GitHub Secrets, while the role name is stored in GitHub Variables?